From: Srivatsa S. Bhat (VMware) <srivatsa@csail.mit.edu>
Date: Tue, 05 Jul 2022 10:00:00 -0700
Subject: [PATCH] Do not install auxiliary module from ice driver sources

Both iavf-4.4.2 and ice-1.8.3 driver sources include a backported
kernel module called auxiliary.ko, which they both depend on. Since we
just need one copy of auxiliary.ko in the final rpm, install it only
from the iavf driver sources and skip the installation from the ice
driver.

Signed-off-by: Srivatsa S. Bhat (VMware) <srivatsa@csail.mit.edu>
---

diff -Naurp ice-1.8.3-orig/src/common.mk ice-1.8.3/src/common.mk
--- ice-1.8.3-orig/src/common.mk	2022-07-05 09:55:22.525371829 -0700
+++ ice-1.8.3/src/common.mk	2022-07-05 10:06:50.421396430 -0700
@@ -416,31 +416,31 @@ endif # check_aux_bus exists
 
 # The out-of-tree auxiliary module we ship should be moved into this
 # directory as part of installation.
-export INSTALL_AUX_DIR ?= updates/drivers/net/ethernet/intel/auxiliary
+# export INSTALL_AUX_DIR ?= updates/drivers/net/ethernet/intel/auxiliary
 
 # If we're installing auxiliary bus out-of-tree, the following steps are
 # necessary to ensure the relevant files get put in place.
-ifeq (${NEED_AUX_BUS},2)
-define auxiliary_post_install
-	install -D -m 644 Module.symvers ${INSTALL_MOD_PATH}/lib/modules/${KVER}/extern-symvers/auxiliary.symvers
-	install -d ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}
-	mv -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_MOD_DIR}/auxiliary.ko \
-	      ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}/auxiliary.ko
-	install -D -m 644 linux/auxiliary_bus.h ${INSTALL_MOD_PATH}/${KSRC}/include/linux/auxiliary_bus.h
-endef
-else
-auxiliary_post_install =
-endif
+# ifeq (${NEED_AUX_BUS},2)
+# define auxiliary_post_install
+# 	install -D -m 644 Module.symvers ${INSTALL_MOD_PATH}/lib/modules/${KVER}/extern-symvers/auxiliary.symvers
+# 	install -d ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}
+# 	mv -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_MOD_DIR}/auxiliary.ko \
+# 	      ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}/auxiliary.ko
+# 	install -D -m 644 linux/auxiliary_bus.h ${INSTALL_MOD_PATH}/${KSRC}/include/linux/auxiliary_bus.h
+# endef
+# else
+# auxiliary_post_install =
+# endif
 
-ifeq (${NEED_AUX_BUS},2)
-define auxiliary_post_uninstall
-	rm -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/extern-symvers/auxiliary.symvers
-	rm -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}/auxiliary.ko
-	rm -f ${INSTALL_MOD_PATH}/${KSRC}/include/linux/auxiliary_bus.h
-endef
-else
-auxiliary_post_uninstall =
-endif
+# ifeq (${NEED_AUX_BUS},2)
+# define auxiliary_post_uninstall
+# 	rm -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/extern-symvers/auxiliary.symvers
+# 	rm -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}/auxiliary.ko
+# 	rm -f ${INSTALL_MOD_PATH}/${KSRC}/include/linux/auxiliary_bus.h
+# endef
+# else
+# auxiliary_post_uninstall =
+# endif
 
 ######################
 # Kernel Build Macro #
diff -Naurp ice-1.8.3-orig/src/Makefile ice-1.8.3/src/Makefile
--- ice-1.8.3-orig/src/Makefile	2022-07-05 09:55:22.521371829 -0700
+++ ice-1.8.3/src/Makefile	2022-07-05 10:08:11.993399348 -0700
@@ -141,7 +141,7 @@ DDP_PKG_DEST := ${DDP_PKG_DEST_PATH}/${D
 DDP_PKG_LINK := ${DRIVER}.pkg
 modules_install: all
 	+$(call kernelbuild,modules_install)
-	${auxiliary_post_install}
+#	${auxiliary_post_install}
 ifneq (${DDP_PKG_ORIGIN},)
 	@# Install DDP package file if provided/required with ice module
 	install -D -m 644 ${DDP_PKG_ORIGIN} ${DDP_PKG_DEST}
@@ -166,7 +166,7 @@ install: modules_install mandocs_install
 # build root. Thus, it must not update the initramfs image or run depmod.
 modules_uninstall:
 	rm -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_MOD_DIR}/${DRIVER}.ko
-	${auxiliary_post_uninstall}
+#	${auxiliary_post_uninstall}
 ifneq (${DDP_PKG_ORIGIN},)
 	@# Uninstall DDP package file if provided/required with ice module
 	(cd ${DDP_PKG_DEST_PATH} && \
